server/issue_test.go

```diff
@@ -279,6 +280,255 @@ func TestRouteShareIssuePublicly(t *testing.T) {
+		expectedErr error
+	}{
+		"No Field": {
+			p: &Plugin{
+				instanceStore: mockInstanceStore{},
+				userStore:     mockUserStore{},
+			},
+			instanceID: testInstance1.InstanceID,
+			client:     nil,
+			connection: nil,
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{},
+				},
+			},
+			expectedErr: nil,
+		},
+		"Success": {
+			p: &Plugin{
+				instanceStore: mockInstanceStore{},
+				userStore:     mockUserStore{},
+			},
+			instanceID: testInstance1.InstanceID,
+			client:     testClient{},
+			connection: &Connection{
+				Settings: &ConnectionSettings{
+					Notifications:          true,
+					RolesForDMNotification: nil,
+				},
+			},
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{
+						Fields: &jira.IssueFields{
+							Creator: &jira.User{},
+						},
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			client, connection, error := tt.wh.fetchConnectedUser(
+				tt.p,
+				tt.instanceID,
+			)
+			assert.Equal(t, tt.connection, connection)
+			assert.Equal(t, tt.client, client)
+			if tt.expectedErr != nil {
+				assert.Error(t, tt.expectedErr, error)
+			}
+		})
+	}
+}
+
+func TestApplyReporterNotification(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+
+	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+	p := Plugin{}
+	p.SetAPI(api)
+	p.instanceStore = p.getMockInstanceStoreKV(1)
+	p.userStore = getMockUserStoreKV()
+
+	tests := map[string]struct {
+		wh         *webhook
+		instanceID types.ID
+		reporter   *jira.User
+	}{
+		"No reporter": {
+			wh: &webhook{
+				eventTypes: map[string]bool{
+					createdCommentEvent: true,
+				},
+				JiraWebhook: &JiraWebhook{
+					Comment: jira.Comment{
+						UpdateAuthor: jira.User{},
+					},
+					Issue: jira.Issue{
+						Key: "TEST-10",
+						Fields: &jira.IssueFields{
+							Type: jira.IssueType{
+								Name: "Story",
+							},
+							Summary: "",
+						},
+						Self: "Abc",
+					},
+				},
+				notifications: []webhookUserNotification{},
+			},
+			instanceID: testInstance1.InstanceID,
+			reporter:   nil,
+		},
+		"Success": {
+			wh: &webhook{
+				eventTypes: map[string]bool{
+					createdCommentEvent: true,
+				},
+				JiraWebhook: &JiraWebhook{
+					Comment: jira.Comment{
+						UpdateAuthor: jira.User{},
+					},
+					Issue: jira.Issue{
+						Key: "TEST-10",
+						Fields: &jira.IssueFields{
+							Type: jira.IssueType{
+								Name: "Story",
+							},
+							Summary: "",
+						},
+						Self: "Abc",
+					},
+				},
+				notifications: []webhookUserNotification{},
+			},
+			instanceID: testInstance1.InstanceID,
+			reporter:   &jira.User{},
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			l1 := len(tt.wh.notifications)
+			p.applyReporterNotification(
+				tt.wh,
+				tt.instanceID,
```

Comment:
Here you can create more sub-test cases like when instance id nil think of more scnerios

Reply:
We cannot initialize instanceID as `nil` because ` We cannot use nil as types.ID value in struct literal`, instead made an another tc in which in which instanceID is initialized with `"initialize"`